src/server/auth/server/api_server.go

```diff
@@ -362,6 +362,48 @@ func (a *apiServer) Activate(ctx context.Context, req *auth.ActivateRequest) (re
 	return &auth.ActivateResponse{PachToken: pachToken}, nil
 }
 
+// RotateToken implements the protobuf auth.RotateToken RPC
+func (a *apiServer) RotateAuthToken(ctx context.Context, req *auth.RotateAuthTokenRequest) (resp *auth.RotateAuthTokenResponse, retErr error) {
+	pachClient := a.env.GetPachClient(ctx)
+	ctx = pachClient.Ctx() // copy auth information
+	// We don't want to actually log the request/response since they contain credentials.
+	defer func(start time.Time) { a.LogResp(nil, nil, retErr, time.Since(start)) }(time.Now())
+
+	// Determine the subject of the request
+	var subject string
+	if req.Subject != "" {
+		subject = req.Subject
+	} else {
+		tokenInfo, err := a.getAuthenticatedUser(ctx)
+		if err != nil {
+			return nil, err
+		}
+		subject = tokenInfo.Subject
+	}
+
+	// First revoke this subject's existing auth_tokens
+	a.RevokeAuthTokensForUser(ctx, &auth.RevokeAuthTokensForUserRequest{Username: subject})
+
+	// If the token hash was in the request, use it and return an empty response.
+	// Otherwise generate a new random token.
+	pachToken := req.PachToken
+	if pachToken == "" {
+		pachToken = uuid.NewWithoutDashes()
+	}
+
+	var err error
+	if req.TTL > 0 {
```

Comment:
I'm conflicted as to whether to expose TTL in the command. On one hand it makes sense to give the admin the power to control the TTL of a new token for a user. On the other by carelessly omitting a TTL value, they grant the token indefinite lifetime.

We could instead have the default value be a Standard length of time, and to declare indefinite lifetime, we could pass another flag, or pass a negative value.

What are your thoughts here @actgardner ?